src/azure-cli/azure/cli/command_modules/role/_params.py

```diff
@@ -181,30 +185,41 @@ def load_arguments(self, _):
-        c.argument('password', options_list=['--password', '-p'], arg_group='Credential',
-                   help="If missing, CLI will generate a strong password")
-
-    with self.argument_context('ad app credential reset') as c:
-        c.argument('cert', arg_group='Credential', validator=validate_cert, help='Certificate to use for credentials')
-        c.argument('years', type=int, default=None, arg_group='Credential', help='Number of years for which the credentials will be valid')
-        c.argument('append', action='store_true', help='Append the new credential instead of overwriting.')
-        c.argument('display_name', help="Friendly name for the password or key.")
-
-        c.argument('create_cert', action='store_true', arg_group='keyCredentials', help='Create a self-signed certificate to use for the credential')
-        c.argument('keyvault', arg_group='keyCredentials', help='Name or ID of a KeyVault to use for creating or retrieving certificates.')
+                       help="Finer grain of expiry time if '--years' is insufficient, e.g. '2020-12-31T11:59:59+00:00' "
+                            "or '2299-12-31'")
+
+            # keyCredential arguments
+            c.argument('cert', arg_group='keyCredential', validator=validate_cert,
+                       help='Certificate to use for credentials. When used with `--keyvault,`, indicates the name of the '
+                            'cert to use or create. Otherwise, supply a PEM or DER formatted public certificate string. '
+                            'Use `@{path}` to load from a file. Do not include private key info.')
+            c.argument('create_cert', arg_group='keyCredential', action='store_true',
+                       help='Create a self-signed certificate to use for the credential. Only the current OS user has '
+                            'read/write permission to this certificate. Use with `--keyvault` to create the certificate in '
+                            'Key Vault. Otherwise, a certificate will be created locally.')
+            c.argument('keyvault', arg_group='keyCredential',
+                       help='Name or ID of a KeyVault to use for creating or retrieving certificates.')
+
+    # --display-name in `ad sp create-for-rbac` is for the sp, not credential, so we only apply it to `credential reset`
+    # commands.
+    for item in ['ad app credential reset', 'ad sp credential reset']:
+        with self.argument_context(item) as c:
+            c.argument('display_name', arg_group='Credential',
+                       help="Friendly name for the password or certificate credential.")
+
+    for item in ['ad app credential list', 'ad app credential delete',
+                 'ad sp credential list', 'ad sp credential delete']:
+        with self.argument_context(item) as c:
+            c.argument('key_id', help='credential key id')
+            c.argument('cert', action='store_true', help='Operate on certificate credentials')
```

Comment:
The original help "a certificate based credential" is confusing in `list` commands.